pkg/controllers/nodeclaim/disruption/controller.go

```diff
@@ -110,14 +112,30 @@ func (c *Controller) Name() string {
 	return "nodeclaim.disruption"
 }
 
+// NodePoolEventHandler is a watcher on v1beta1.NodeClaim that maps Provisioner to NodeClaims based
+// on the v1beta1.NodePoolLabelKey and enqueues reconcile.Requests for the NodeClaim
+func NodePoolEventHandler(c client.Client) handler.EventHandler {
```

Comment:
Not sure that we need to move this into here in this PR. Let's keep the event handlers in one place in the `nodeclaimutil` package for now